service/matching/physical_task_queue_manager.go

```diff
@@ -154,17 +257,19 @@ func newPhysicalTaskQueueManager(
 		metrics.OperationTag(metrics.MatchingTaskQueueMgrScope),
 		metrics.WorkerBuildIdTag(queue.VersionSet()))
 	pqMgr := &physicalTaskQueueManagerImpl{
-		status:               common.DaemonStatusInitialized,
-		partitionMgr:         partitionMgr,
-		namespaceRegistry:    e.namespaceRegistry,
-		matchingClient:       e.matchingRawClient,
-		metricsHandler:       e.metricsHandler,
-		clusterMeta:          e.clusterMeta,
-		queue:                queue,
-		logger:               logger,
-		throttledLogger:      throttledLogger,
-		config:               config,
-		taggedMetricsHandler: taggedMetricsHandler,
+		status:                     common.DaemonStatusInitialized,
+		partitionMgr:               partitionMgr,
+		namespaceRegistry:          e.namespaceRegistry,
+		matchingClient:             e.matchingRawClient,
+		metricsHandler:             e.metricsHandler,
+		clusterMeta:                e.clusterMeta,
+		queue:                      queue,
+		logger:                     logger,
+		throttledLogger:            throttledLogger,
+		config:                     config,
+		taggedMetricsHandler:       taggedMetricsHandler,
+		TasksAddedInIntervals:      newTaskTracker(clock.NewRealTimeSource(), intervalSize, totalIntervalSize),
+		TasksDispatchedInIntervals: newTaskTracker(clock.NewRealTimeSource(), intervalSize, totalIntervalSize),
```

Comment:
My two cents:

We could keep them simpler, but I think it could be better to have `Intervals` in the name here since the whole purpose is to track the number of tasks in a *fixed interval*. Moreover, `tasksAdded` also makes it seem as if we are storing all the tasks which are being added which is not true as after a given interval, flushing of the buffer takes place. Thoughts?

Reply:
Yeah, maybe "tasksAddedInWindow" or "recentTasksAdded" or something like that?

But even if you keep the original names, please make them unexported